### ISSUE TITLE
immediate tasks

### ISSUE THREAD

**Issue by jhconning** [jhconning/FarmSizeView#5]
- [ ]  think of lognormal fit
- [ ]  add legends to plots

**Comment by jhconning** [comment]
@yqlingc  I've updated the distn_fit.ipynb to illustrate MLE on binned data.... Seems to work.

**Comment by yqlingc** [comment]
Fantastic! I will download the new file to see the new data illustration. Thank you very much.

________________________________

**Comment by mbaker21231** [comment]
Jonathan --

Well done; that is a neat way of doing it. I wonder if there are advantages to doing it that way rather than matching moments...

Matt

**Comment by jhconning** [comment]
Yeah for simple examples like this normal much  easier to match moments (the sample mean and std are the MLE estimators)... This is just a first step to warm up to estimate a lognormal with censored data (or other likelihood functions). All this only because I noticed Adamopolos and Restuccia seem to have fitted their model to censored data:

Note how 54% of the land in use is in their top bin category '>809 has'.  
```
classlabel = ['<4', '4-20', '20-28', '28-40', '40-57', '57-73', '73-89','89-105','105-202', '202-405', '405-809', '809+'] 
farm_pdf_data = [.1056, .2813, .0698, .0871, .0794, .0633, .0397, .0310, .0964, .0679, .0420, .0365]
land_pdf_data=[.0012, .0173, .0097, .0171, .0220, .0238, .0187, .0176, .0823, .1129, .1384, .5389 ]
```
 I'll soon get to a simple censored example

**Comment by jhconning** [comment]
The newest version of notebook just posted takes care of the censoring in that last bin (via MLE).   Would be better to adapt this to using lognormal ....   picked up a few tricks on using numpy, plt.hist and scipy.optimize.minimize along the way that will come in handy..  

Need to go back and see how/whether A&R  dealt with that top bin issue in their larger program.

**Comment by mbaker21231** [comment]
Interesting. Another idea might be to consider more general distributions and maybe even some that have "fat tails." For example: [The Burr distribution](https://en.wikipedia.org/wiki/Burr_distribution). I wonder what existing methods are like to test the goodness-of-fit obtained by different distributions.

Matt

**Comment by jhconning** [comment]
in the latest commit 92cfb15909970bf076d74fc34423a70d55cc19d7
1. I'm now fitting to a censored lognormal. 
2. I'm also fitting a censored Burr distribution

  I'm getting essentially the same parameter estimates in either case which seems to suggest the more generalized shape of the Burr is not adding anything extra in this case. (?)  

Here's a plot of the fitted censored lognormal. 
![image](https://cloud.githubusercontent.com/assets/5048607/24132432/ea143c80-0dcb-11e7-8d27-0ab62c2dd50a.png)

**Comment by jhconning** [comment]
I've added lognormal and Burr fits for the **land-pdf-data* (where it makes most sense to use censored).  The estimates don't seem stable to changes in the initial starting guess.  

Another problem I've noticed is that the histograms aren't quite right..  We can't use the usual plt.hist because we're using pre-counted data (see [this]((http://stackoverflow.com/questions/19212508/plotting-a-histogram-from-pre-counted-data-in-matplotlib) so use the weights option to pass the counts. But something is wrong as some bin counts are not showing and all counts seem to be plotted on upper edge of bins.  Setting bins=12 doesn't seem to help.
 
![image](https://cloud.githubusercontent.com/assets/5048607/24146386/dccb9592-0e0b-11e7-9ee2-37f6575c22f4.png)

Back to the fit issue, looking at this (imperfect as it is ) plot, could part of the problem be that the data is somewhat bimodal (two humped)...

But rather than press further on MLE methods in python... I think it's wiser to fall back to a simpler matching of moments approach (either with with adjustment for censored data or better yet and contra A&R, let's just get more bin data from the US Ag Census; I never understood why they limited themselves to less).

**Comment by jhconning** [comment]
Fix of sorts to histogram plotting problem mentioned above is to plot frequencies at midpoint of bins and also set bins=60 (a high number) to help make sure pyplot gets them all. For example:

![image](https://cloud.githubusercontent.com/assets/5048607/24147973/1c7b674c-0e13-11e7-9f4c-175078fbdbd4.png)
